calculator_test.go

```diff
@@ -2,23 +2,407 @@ package calculator_test
+		want        float64
+		description string
+		errExpected bool
+	}{
+		{
+			description: "dividing by zero",
+			a:           2,
+			b:           0,
+			want:        123456789,
+			errExpected: true,
+		},
+		{
+			description: "two positive numbers whos quotient is positive",
+			a:           20,
+			b:           2,
+			want:        10,
+		},
+		{
+			description: " a positive and negative number whos quotient is negative",
+			a:           10,
+			b:           -2,
+			want:        -5,
+		},
+		{
+			description: "a positive decimal and negative decimal whos quotient is a negative decimal",
+			a:           8.4,
+			b:           -2.5,
+			want:        -3.3600000000000003,
+		},
+	}
+
+	t.Parallel()
+
+	for _, c := range testCases {
+		err, got := calculator.Divide(c.a, c.b)
+		if err != nil && c.errExpected == false {
+			t.Errorf("error received while testing %s. The function call was: Divide(%v, %v), and the error was: %v", c.description, c.a, c.b, err)
+		}
+
+		// Only fail on got != want if an error was not expected
+		if c.errExpected == false && got != c.want {
+			t.Errorf("want %v, got %v, while testing %s. The function call was: Divide(%v, %v)", c.want, got, c.description, c.a, c.b)
+		}
+	}
+}
+
+// Randomly test the variadic call to Divide()
+func TestDivideVariadicRandomly(t *testing.T) {
+	t.Parallel()
+	rand.Seed(time.Now().UnixNano())
+
+	// rand.Float64() returns a number in 0.0-1.0
+	// Use another randomly-generated int to vary the whole number.
+	a := rand.Float64() * float64(rand.Intn(500))
+	b := rand.Float64() * float64(rand.Intn(500))
+
+	// Avoid dividing by 0
+	if b == 0 {
+		b += 0.1
+	}
+
+	// v will be assigned multiple random float64 in a slice
+	v := randFloat64Slice(5)
+
+	// Get our own quotient to compare to Divide()
+	// Also change any randomly-generated 0.0s, to avoid dividing-by-zero!
+	var want float64 = a / b
+	for i, x := range v {
+		if x == 0 {
+			x += 0.1
+			v[i] = x // change the slice
+		}
+		want = want / x
+	}
+
+	err, got := calculator.Divide(a, b, v...)
+	if err != nil {
+		t.Errorf("error received while testing a random variadic case. The function call was: Divide(%v, %v, %v), and the error was: %v", a, b, v, err)
+	}
+
+	t.Logf("Random variadic test: Divide(%v, %v, %v), wants %v, got %v", a, b, v, want, got)
+	if got != want {
+		t.Errorf("want %v, got %v, while testing a random variadic case. The function call was: Divide(%v, %v, %v)", want, got, a, b, v)
+	}
+}
+
+func TestSqrt(t *testing.T) {
+	// Define test cases
+	testCases := []struct {
+		a           float64
+		b           float64
+		want        float64
+		description string
+		errExpected bool
+	}{
+		{
+			description: "negative input",
+			a:           -64,
+			want:        123456789,
+			errExpected: true,
+		},
+		{
+			description: "64",
+			a:           64,
+			want:        8,
+		},
+	}
+
+	t.Parallel()
+
+	for _, c := range testCases {
+		err, got := calculator.Sqrt(c.a)
+		if err != nil && c.errExpected == false {
```

Comment:
A short way to write `if X == false` is `if !X`, isn't it?

Reply:
I resisted this initially for super-clarity and my not hearing punctuation by default when using a screen-reader. That doesn't mean I can't examine punctuation when needed, and I do like the shorter form. :)